snowplow-tracker/src/main/java/com/snowplowanalytics/core/ecommerce/EcommerceControllerImpl.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2015-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.core.ecommerce
+
+import androidx.annotation.RestrictTo
+import com.snowplowanalytics.core.constants.Parameters
+import com.snowplowanalytics.core.constants.TrackerConstants
+import com.snowplowanalytics.core.tracker.ServiceProviderInterface
+import com.snowplowanalytics.snowplow.configuration.PluginConfiguration
+import com.snowplowanalytics.snowplow.ecommerce.EcommerceController
+import com.snowplowanalytics.snowplow.ecommerce.entities.Product
+import com.snowplowanalytics.snowplow.ecommerce.entities.Promotion
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+class EcommerceControllerImpl(val serviceProvider: ServiceProviderInterface) : EcommerceController {
+
+    override fun setEcommerceScreen(type: String, language: String?, locale: String?) {
+        val plugin = PluginConfiguration("ecommercePageTypePluginInternal")
```

Comment:
Should this entity be added to all events or just the `snowplow_ecommerce_action`? You could use the filter parameter in the `PluginConfiguration` to provide a list of schemas to which the entity should be added.

(the same comment for the `setEcommerceUser`).

Reply:
I agree that it is surprising that it's added to all the events, not just the ecommerce ones. However, this is the behaviour in the JS tracker so I think it's best to keep it the same.